frontend/src/flux/reducers/authReducer.js

```diff
@@ -0,0 +1,61 @@
+import {
+  AUTH_ERROR,
+  LOGIN_FAIL,
+  LOGIN_SUCCESS,
+  LOGOUT_SUCCESS,
+  REGISTER_FAIL,
+  REGISTER_SUCCESS,
+  USER_LOADED,
+  USER_LOADING,
+} from '../actions/types';
+
+const initialState = {
+  token: localStorage.getItem('token'),
+  isAuthenticated: null,
+  isLoading: false,
+  user: null,
+};
+
+export default function (state = initialState, action) {
+  switch (action.type) {
+    case USER_LOADING:
+      return {
+        ...state,
+        isLoading: true,
+      };
+
+    case USER_LOADED:
+      return {
+        ...state,
+        isAuthenticated: true,
+        isLoading: false,
+        user: action.payload,
+      };
+
+    case LOGIN_SUCCESS:
+    case REGISTER_SUCCESS:
+      localStorage.setItem('token', action.payload.token);
+      return {
+        ...state,
+        ...action.payload, // don't need to specify user and token because they are included in the payload TODO: delete this comment later
```

Comment:
@marcianszlovak that's true, but on the frontend we don't want to look into the jwt's content